indexing-service/src/main/java/org/apache/druid/indexing/common/TaskRealtimeMetricsMonitorBuilder.java

```diff
@@ -45,15 +47,20 @@ public static RealtimeMetricsMonitor build(Task task, FireDepartment fireDepartm
     );
   }
 
-  public static TaskRealtimeMetricsMonitor build(Task task, FireDepartment fireDepartment, RowIngestionMeters meters)
+  public static TaskRealtimeMetricsMonitor build(
+      Task task,
+      FireDepartment fireDepartment,
+      RowIngestionMeters meters
+  )
   {
     return new TaskRealtimeMetricsMonitor(
         fireDepartment,
         meters,
         ImmutableMap.of(
             DruidMetrics.TASK_ID, new String[]{task.getId()},
             DruidMetrics.TASK_TYPE, new String[]{task.getType()}
-        )
+            ),
+        (Map<String, Object>) task.getContext().get("metricMetadata")
```

Comment:
this pattern is used in other places in the code, but if metricMetadata is not a map, I think this will crash. Should we introduce a new pattern where this just ignores the metricMetadata object if it is not a map or is failing loudly the desired behavior?

Reply:
`metricMetadata` is the key users would need to pass in the context map to add additional labels to the ingestion metadata. Any thoughts on a better name?

Reply:
"tags".  On your blowing up comment, write defensive code please.